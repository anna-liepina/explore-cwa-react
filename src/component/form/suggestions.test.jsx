import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Suggestions from './suggestions';

describe('<Suggestions/>', () => {
    const props = {
        options: [
            {
                label: 'props.label 1',
                value: 1,
            },
            {
                label: 'props.label 2 with hierarchy',
                hierarchy: ['{{root}}', '{{sub-children}}', '{{label-with-hierarchy}}'],
                value: 2,
            },
            {
                label: 'props.label 3 with description',
                description: '{{description}}',
                value: 'three',
            },
        ],
    };
    const optionalProps = {
        'data-cy': 'optProps.data-cy',
        className: 'optProps.className',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment, debug } = render(<Suggestions {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with optional/required props', () => {
            const { asFragment } = render(<Suggestions {...props} {...optionalProps} />);

            expect(asFragment()).toMatchSnapshot();
        });

        describe('specific optional props', () => {
            it(`[::hashmap] - should NOT render options with values which are present in ::hashmap`, () => {
                const { asFragment } = render(<Suggestions {...props} hashmap={{ 2: true, three: true }} />);
    
                expect(asFragment()).toMatchSnapshot();
            });    
        })
    });
});
