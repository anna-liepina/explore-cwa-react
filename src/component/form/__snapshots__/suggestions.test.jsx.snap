// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Suggestions/> render specific optional props [::hashmap] - should NOT render options with values which are present in ::hashmap 1`] = `
<DocumentFragment>
  <div
    class="suggestions "
  >
    <div
      class="suggestion "
      data-cy="-suggestion-0"
      data-id="0"
    >
      <div
        class="suggestion__label"
        data-id="0"
      >
        props.label 1
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Suggestions/> render with default/required props 1`] = `
<DocumentFragment>
  <div
    class="suggestions "
  >
    <div
      class="suggestion "
      data-cy="-suggestion-0"
      data-id="0"
    >
      <div
        class="suggestion__label"
        data-id="0"
      >
        props.label 1
      </div>
    </div>
    <div
      class="suggestion "
      data-cy="-suggestion-1"
      data-id="1"
    >
      <div
        class="suggestion__hierarchy"
      >
        <span
          class="suggestion__hierarchy-label "
          data-id="1"
        >
          {{root}}
        </span>
        <span
          class="suggestion__hierarchy-label "
          data-id="1"
        >
          {{sub-children}}
        </span>
        <span
          class="suggestion__hierarchy-label "
          data-id="1"
        >
          {{label-with-hierarchy}}
        </span>
      </div>
      <div
        class="suggestion__label"
        data-id="1"
      >
        props.label 2 with hierarchy
      </div>
    </div>
    <div
      class="suggestion "
      data-cy="-suggestion-2"
      data-id="2"
    >
      <div
        class="suggestion__label"
        data-id="2"
      >
        props.label 3 with description
      </div>
      <p
        class="suggestion__description"
        data-id="2"
      >
        {{description}}
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Suggestions/> render with optional/required props 1`] = `
<DocumentFragment>
  <div
    class="suggestions optProps.className"
  >
    <div
      class="suggestion "
      data-cy="optProps.data-cy-suggestion-0"
      data-id="0"
    >
      <div
        class="suggestion__label"
        data-id="0"
      >
        props.label 1
      </div>
    </div>
    <div
      class="suggestion "
      data-cy="optProps.data-cy-suggestion-1"
      data-id="1"
    >
      <div
        class="suggestion__hierarchy"
      >
        <span
          class="suggestion__hierarchy-label "
          data-id="1"
        >
          {{root}}
        </span>
        <span
          class="suggestion__hierarchy-label "
          data-id="1"
        >
          {{sub-children}}
        </span>
        <span
          class="suggestion__hierarchy-label "
          data-id="1"
        >
          {{label-with-hierarchy}}
        </span>
      </div>
      <div
        class="suggestion__label"
        data-id="1"
      >
        props.label 2 with hierarchy
      </div>
    </div>
    <div
      class="suggestion "
      data-cy="optProps.data-cy-suggestion-2"
      data-id="2"
    >
      <div
        class="suggestion__label"
        data-id="2"
      >
        props.label 3 with description
      </div>
      <p
        class="suggestion__description"
        data-id="2"
      >
        {{description}}
      </p>
    </div>
  </div>
</DocumentFragment>
`;
